bot/exts/utils/snekbox.py

```diff
@@ -326,17 +391,81 @@ async def eval_command(self, ctx: Context, *, code: str = None) -> None:
 
         while True:
             self.jobs[ctx.author.id] = datetime.datetime.now()
-            code = self.prepare_input(code)
             try:
-                response = await self.send_eval(ctx, code)
+                response = await self.send_eval(ctx, code, args=args, format_func=format_func)
             finally:
                 del self.jobs[ctx.author.id]
 
-            code = await self.continue_eval(ctx, response)
+            code, args = await self.continue_eval(ctx, response, ctx.command)
             if not code:
                 break
             log.info(f"Re-evaluating code from message {ctx.message.id}:\n{code}")
 
+    async def format_timeit_output(self, output: str) -> tuple[str, str]:
+        """
+        Parses the time from the end of the output given by timeit.
+
+        If an error happened, then it won't contain the time and instead proceed with regular formatting.
+        """
+        split_output = output.rstrip("\n").rsplit("\n", 1)
+        if len(split_output) == 2 and TIMEIT_OUTPUT_REGEX.fullmatch(split_output[1]):
+            return split_output[1], None
+
+        return await self.format_output(output)
```

Comment:
To continue on the resolved conversation, why is this necessary at all? 

If the command runs properly, the output should be a single line we can directly send anyway. While passing it through format_output is a bit unnecessary, the changeable format func does complicate things considerably compared to just not doing it

From my understanding that is what's happening anyway as the split_output will only contain a single element if it's the timeit stdout